src/Client/ClientBase.cpp

```diff
@@ -564,7 +564,16 @@ try
         if (!pager.empty())
         {
             if (SIG_ERR == signal(SIGPIPE, SIG_IGN))
-                throwFromErrno("Cannot set signal handler.", ErrorCodes::CANNOT_SET_SIGNAL_HANDLER);
+                throwFromErrno("Cannot set signal handler for SIGPIPE.", ErrorCodes::CANNOT_SET_SIGNAL_HANDLER);
+            /// We need to reset signals that had been installed in the
+            /// setupSignalHandler() since terminal will send signals to both
+            /// processes and so signals will be delivered to the
+            /// clickhouse-client/local as well, which will be terminated when
+            /// signal will be delivered second time.
+            if (SIG_ERR == signal(SIGINT, SIG_IGN))
```

Comment:
If you ignore signals, how can you cancel a query that's being run?

For example, if I do `--pager less` send a neverending query (`Select * from system.numbers`) and go to the end (`G command`) now I can't cancel the query from the  CLI. Same if you use `less +F`, you stop being able to stop the query at all as ctrl+c does nothing.

Reply:
Makes sense. With `output_format_pretty_max_rows` disabled it seems to be working fine. I'm not sure if I'd expect  Ctrl+C to also stop the source but I guess not, and doing it on pager exit seems the most reasonable way forward.